packages/shopping/src/controllers/user-management.controller.ts

```diff
@@ -297,4 +306,70 @@ export class UserManagementController {
 
     return {token};
   }
+
+  @post('/users/reset-password/init', {
+    responses: {
+      '200': {
+        description: 'Confirmation that reset password email has been sent',
+      },
+    },
+  })
+  async resetPasswordInit(
+    @requestBody() resetPasswordInit: ResetPasswordInit,
+  ): Promise<string> {
+    if (!isemail.validate(resetPasswordInit.email)) {
+      throw new HttpErrors.UnprocessableEntity('Invalid email address');
+    }
+
+    const nodeMailer: NodeMailer = await this.userManagementService.requestPasswordReset(
+      resetPasswordInit.email,
+    );
+
+    if (nodeMailer.accepted.length) {
+      return 'Successfully sent reset password link';
+    }
+    throw new HttpErrors.InternalServerError(
+      'Error sending reset password email',
+    );
+  }
+
+  @put('/users/reset-password/finish', {
+    responses: {
+      '200': {
+        description: 'A successful password reset response',
+      },
+    },
+  })
+  async resetPasswordFinish(
+    @requestBody() keyAndPassword: KeyAndPassword,
+  ): Promise<string> {
+    validateKeyPassword(keyAndPassword);
+
+    const foundUser = await this.userRepository.findOne({
+      where: {resetKey: keyAndPassword.resetKey},
```

Comment:
Since `resetKey` defaults to "empty string" for newly created users and those that just reset their password, by making a request to `/users/reset-password/finish?resetKey=`, the account of the first user with an empty `resetKey` can be take over.

Once that account is taken over, we can initiate a password reset for that user to set the `resetKey` value for this user. A request to `/users/reset-password/finish?resetKey=` then will then yield the next user, etc.

Solution: always do `model.resetKey = uuidv4()`.

Reply:
Oh, as I scrolled down, I could see empty `resetKey` is being checked. Please disregard the earlier comment.

Reply:
@hacksparrow but we enforce a check for non empty reset key to avoid such a scenario on the validation